### ISSUE TITLE
One request: regarding input composition in Korean letters

### ISSUE THREAD

**Issue by khsbory** [nvaccess/nvda#5964]
Hello, I'm Hyongsop Kim in Korea and NVDA user.
In case of English, there is no input composition mode when I type characters, but in case of Korea, input composition is appeared.
I understand that in case of Asian characters, input composition is needed.
By the way, how about adding input composition on or off function even Asian characters?
Because, in case of hangul (Korean character), there is no candidate needed.
Also once input composition is started, I can't review whole sentence.
This is very critical in Korean, because fast text review is impossible because of composition mode is on.
Please chechk this issue.

Thank you.

**Comment by dnz3d4c** [comment]
@josephsl I think This issue is very critical. Please check issue.

**Comment by josephsl** [comment]
Hi, I’ll look into this – please have Mr. Kim write to NVDA Korea FB group in Korean describing this issue so we can help others investigate. Thank you.

**Comment by khsbory** [comment]
Hello, again.
I'm Hyongsop Kim in Korea.
Regarding input composition and Korean character typing issue, I want to let you know one additional information.
In some editing area, if I type Korean characters with Nvda, some characters are separated to each  consonant vowel or consonant vowel and final consonant.
Korean's each character has one consonant, one vowel and one final consonant.
Whether including final consonant is based on each character.
However first consonant and vowel is required at least.
By the way, if I type one character that have one first consonant, one vowel and one final consonant, each one has separated and not complete with one character.
As I said before, this issue is not happened in all editing area, but in some editing area, the problem is occurred.
Therefore I really hope this should be fixed for Korean Nvda users.
If you need additional information, please let me know.

Thank you.

**Comment by feerrenrut** [comment]
@khsbory Could you please give us an example of an editing area in a program that has this behaviour?@josephsl can I assign this issue to you?

**Comment by khsbory** [comment]
Thank you for your concern.
It depends on windows circumstance.
In other words, now in Windows 10, in Skype, this behavior is occurred.
However in prior Windows 10, in Skype, there was no issue, but in Firefox nightly build, the issue was occurred such as ckeditor area.
The important thing is regardless of Windows configuration is different, the problem is occurred in some editing area.
The program is different depends on Windows configuration.
I couldn't find what will affect this issue?
Lastly, because of input composition error, the problem may occurred.
In other words, if I type one character with first consonant, vowel and final consonant, the input composition doesn't work even if it is Korean character.

Thank you.

**Comment by josephsl** [comment]
Hi,

Although I'm sort of knowledgeable in that regard (I was born in Korea), this is something @MichaelDCurran and @Nishmotz knows better. But to the issue at hand:

I sort of see what's up... But to test what I think is going on (composition string being announced), may I ask you to try:

1. Open an edit field and type something in Korean.
2. Open NVDA menu/Preferences/Input composition.
3. Uncheck "report changes to the composition string" and press ENTER.
4. Try typing something in Korean.

Expected: you should not hear changes to the composition string - that is, formation of Korean characters.

in regards to braille display, and reviewing what you wrote in Korean braille, I think it was addressed not long ago.

Background: All CJK characters are pictorial - that is, they are composed of shapes. This is due to the fact that, historically, Chinese, Japanese, and Korean scripts were written in Chinese characters. Then not long ago (several hundred years ago), Korea and Japan created their own writing systems. Because of diverse characters present in CJK writing systems, the bulk of earlier versions of Unicode was devoted to these characters.

A typical input composition mechanism goes something like this:

1. A CJK user switches to their native character input mode (in my case, Korean).
2. User starts typing.
3. For each part of characters typed, IME (Input Method Editor) will fill a box with the character part the user just typed.
4. This cycle repeats until the actual CJK character is created. In case of Chinese and Japanese, for some IME's, a candidate list window appears, allowing the user to select the desired character. In case of Korean, once the IME recognizes the character in question and subsequently entered into the edit field, the box cursor moves on, sort of similar to Latin-based scripts.

I think the last part is where the confusion could come in: because Korean Hangul writing system isn't stroke-based like Chinese or some methods for entering Japanese, a special care should be taken when handling Korean input. In a way, think of it as a hybrid of writing English and Korean, except for cursor movement: in English, cursor moves after typing each symbol, whereas in Korean, cursor moves after the Korean character (Unicode char) is recognized by the IME.

Thanks.

**Comment by khsbory** [comment]
Hello, regarding input composition within Korean typing issue, I'm wondering that will it be fixed.
Even if it is so long time passed, I couldn't find any additional updated issue about it.
Because of NVDA's input composition with Asian characters, Korean nvda users have difficulty typing something with Korean characters, within specific apps.
For example, in Skype desktop application in Windows 10, if I type Korean, all spells are separated.
Korean has consonnance, vowel and/or final consonnance, at least one character has one consonant and one vowel, but each spell is separated in skype application.
You can test it below.
1. Open skype.
2. In contact list, select one friend and press enter.
3. In edit box, set to Korean and native input, type d.
Originally, NVDA should say d selected, of course d is Korean character.
But NVDA just say d, d is separated and is not composed with vowel.
Please consider to fix this issue.

Thank you.

**Comment by dnz3d4c** [comment]
> khsbory commented 15 hours ago
Hello, regarding input composition within Korean typing issue, I'm wondering that will it be fixed.
Even if it is so long time passed, I couldn't find any additional updated issue about it.
Because of NVDA's input composition with Asian characters, Korean nvda users have difficulty typing something with Korean characters, within specific apps.
For example, in Skype desktop application in Windows 10, if I type Korean, all spells are separated.
Korean has consonnance, vowel and/or final consonnance, at least one character has one consonant and one vowel, but each spell is separated in skype application.
You can test it below.
1. Open skype.
2. In contact list, select one friend and press enter.
3. In edit box, set to Korean and native input, type d.
Originally, NVDA should say d selected, of course d is Korean character.
But NVDA just say d, d is separated and is not composed with vowel.
Please consider to fix this issue.
Thank you.

@josephsl This problem is very serious. I do not know exactly when, but the above problem occurred. Korean users using Skype desktop applications on business are complaining of inconvenience. Review please!
Thanks.

**Comment by josephsl** [comment]
Hi, I’m hoping this is something that can be taken care of by this screen reader promotion and development project in Korea (please Dr. Jang for more info). Thanks.

**Comment by bhavyashah** [comment]
@josephsl With reference to https://github.com/nvaccess/nvda/issues/5964#issuecomment-389731723, did this issue get taken care of by the screen reader promotion and development project you were referring to? My understanding of this issue is very limited, but it seems from https://github.com/nvaccess/nvda/issues/5964#issuecomment-220636018 and from https://github.com/nvaccess/nvda/issues/5964#issuecomment-389498574 that it has a significant negative impact on those that it affects, i.e. Korean NVDA users.

**Comment by josephsl** [comment]
Hi, I’m no longer in charge of NVDA promotion and development in Korea, as I have moved on from active translations work. Thanks.

**Comment by Adriani90** [comment]
@khsbory, @ungjinPark, @dnz3d4c could you please give an update on how the behavior is now in NVDA 2022.4 or 2023.1 beta?
